pkg/cmd/admin/router/router.go

```diff
@@ -552,24 +552,46 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 	var clusterIP string
 
 	output := cfg.Action.ShouldPrint()
-	generate := output
+	generateSvc := output
 	service, err := kClient.Services(namespace).Get(name)
 	if err != nil {
-		if !generate {
+		if !generateSvc {
 			if !errors.IsNotFound(err) {
 				return fmt.Errorf("can't check for existing router %q: %v", name, err)
 			}
 			if !output && cfg.Action.DryRun {
```

Comment:
output here is the same as generateSvc, I think, so you can simplify this and remove the !output condition entirely since it was already tested by the !generateSvc above.